### PR TITLE
Gen 3-4: Fix Snatch when used by multiple pokemon

### PR DESCRIPTION
**Pull request by Karthik99999**
According to Bulbapedia and the long description of the move here on showdown, Snatch should be able to steal moves that were called by another user of Snatch.

### CODE REVIEW

File: data/mods/gen4/moves.ts [hunk 1]
```diff
@@ -1433,6 +1433,27 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+		inherit: true,
+		condition: {
+			duration: 1,
+			onStart(pokemon) {
+				this.add('-singleturn', pokemon, 'Snatch');
+			},
+			onAnyPrepareHitPriority: -1,
+			onAnyPrepareHit(source, target, move) {
+				const snatchUser = this.effectState.source;
+				if (snatchUser.isSkyDropped()) return;
```
Comment (DaWoblefet) [inline data/mods/gen4/moves.ts:1446]: This would be unnecessary since it's pre-Sky Drop.

Comment (Karthik99999) [reply data/mods/gen4/moves.ts:1446]: I saw other code check future gen stuff (specifically in the gen 4 moves files, uproar checks for throat chop), which is why I left the future gen stuff in.

Comment (DaWoblefet) [reply data/mods/gen4/moves.ts:1446]: I suppose it's fine if someone uses Sky Drop in Gen 4 custom game or smth. It can be left in.

**Review by Marty-D — APPROVED** [review]
Great, thanks!

**Pull request merged**